iam.tf

```diff
@@ -0,0 +1,107 @@
+data "aws_iam_policy_document" "ec2" {
+  statement {
+    sid     = "EC2AssumeRole"
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+module "ec2_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=0.1.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "ec2"
+}
+
+resource "aws_iam_role" "ec2" {
+  name               = "${module.ec2_label.id}"
+  assume_role_policy = "${data.aws_iam_policy_document.ec2.json}"
+}
+
+module "instance_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=0.1.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "instance"
+}
+
+resource "aws_iam_instance_profile" "instance_profile" {
+  name = "${module.instance_label.id}"
+  role = "${aws_iam_role.ec2.name}"
+}
+
+data "aws_iam_policy_document" "datapipeline_resource" {
+  statement {
+    sid     = "EC2AssumeRole"
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+module "resource_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=0.1.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "resource"
```

Comment:
Does that make sense? Because if we used it the way you wrote it, we could only every have one instance of `tf_efs_backup` for for every (`namespace`, `stage`) combination. We need to have one per (`namespace`, `stage`, `name`) combination.